gflownet/envs/cube.py

```diff
@@ -593,6 +616,34 @@ def _make_increments_distribution(
         beta_distr = Beta(alphas, betas)
         return MixtureSameFamily(mix, beta_distr)
 
+    def _mask_ignored_dimensions(
+        self,
+        mask: TensorType["n_states", "policy_outputs_dim"],
+        tensor_to_mask: TensorType["n_states", "n_dim"],
+    ) -> MixtureSameFamily:
+        """
+        Makes the actions, logprobs or log jacobian entries of ignored dimensions zero.
+
+        Since the shape of all the tensor of actions, the logprobs of increments and
+        the log of the diagonal of the Jacobian must be the same, this method makes no
+        distiction between for simplicity.
+
+        Args
+        ----
+        mask : tensor
+            Boolean mask indicating (True) which dimensions should be set to zero.
+
+        tensor_to_mask : tensor
+            Tensor to be modified. It may be a tensor of actions, of logprobs of
+            increments or the log of the diagonal of the Jacobian.
+        """
+        is_ignored_dim = mask[:, -self.n_dim :]
+        if torch.any(is_ignored_dim):
+            shape_orig = tensor_to_mask.shape
+            tensor_to_mask[is_ignored_dim] = 0.0
+            tensor_to_mask = tensor_to_mask.reshape(shape_orig)
```

Comment:
I'm not sure I understand the purpose of this line. Do it mean that the previous line (L643) changes the shape of `tensor_to_mask`?

Reply:
Yes, it does. It flattens the tensor, annoyingly enough. Is there a more elegant way of handling this?

Reply:
For context: both `is_ignored_dim` and `tensor_to_mask` are 2D tensors of the same shape. `is_ignored_dim` is boolean.